fcrepo-persistence-ocfl/src/main/java/org/fcrepo/persistence/ocfl/OCFLPeristentStorageUtils.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Licensed to DuraSpace under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * DuraSpace licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.fcrepo.persistence.ocfl;
+
+import org.apache.jena.riot.RDFFormat;
+import org.apache.jena.riot.system.StreamRDF;
+import org.fcrepo.kernel.api.RdfStream;
+import org.fcrepo.persistence.api.exceptions.PersistentStorageException;
+import org.fcrepo.persistence.ocfl.api.OCFLObjectSession;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.io.PipedInputStream;
+import java.io.PipedOutputStream;
+
+import static java.lang.String.format;
+import static org.apache.jena.riot.RDFFormat.NTRIPLES;
+import static org.apache.jena.riot.system.StreamRDFWriter.getWriterStream;
+
+/**
+ * A set of utility functions for supporting OCFL persistence activities.
+ *
+ * @author dbernstein
+ * @since 6.0.0
+ */
+public class OCFLPeristentStorageUtils {
+
+    private static final Logger log = LoggerFactory.getLogger(OCFLPeristentStorageUtils.class);
+
+    private OCFLPeristentStorageUtils() {
+    }
+
+    /**
+     * The directory within an OCFL Object's content directory that contains
+     * information managed by Fedora.
+     */
+    public static final String INTERNAL_FEDORA_DIRECTORY = ".fcrepo";
+    /**
+     * The default RDF on disk format
+     * TODO Make this value configurable
+     */
+
+    public static RDFFormat DEFAULT_RDF_FORMAT = NTRIPLES;
+    /**
+     * TODO Make this value configurable
+     * The default extension for the rdf files.
+     */
+    public static String DEFAULT_RDF_EXTENSION = ".n3";
+
+    /**
+     * Resolves the subpath of the resourceId based on the ancestor's resource id.
+     *
+     * @param ancestorResourceId The ancestor resource
+     * @param resourceId         The identifier of the resource whose subpath you wish to resolve.
+     * @return The resolved subpath
+     */
+    public static String resolveSubpath(final String ancestorResourceId, final String resourceId) {
```

Comment:
I'm wondering if it might make sense to name this `relativize` or `relativizeSubpath`.

This is based off the Path and URI classes, which use `resolve` to build up URIs. For example, `Path.get("/some/path").resolve("morestuff")` yields `/some/path/morestuff`. 

Meanwhile `relativize` produces the relative path between a base path and a subpath. So `Paths.get("/some").relativize(Paths.get("/some/path/morestuff"))` gives `path/morestuff'.

I suspect having a `resolve` or `resolveSubpath` method would be useful, for instance it might be used https://github.com/fcrepo4/fcrepo4/pull/1577/files#diff-0e0998a46c3ddf08213ec5158852e9f6R60